src/shared/systems/serialization.hpp

```diff
@@ -0,0 +1,140 @@
+/*
+** EPITECH PROJECT, 2022
+** Manager.hpp
+** File description:
+** .
+*/
+
+#pragma once
+
+#include "../ECS/Components.hpp"
+#include "../ECS/ECS.hpp"
+#include "../ECS/Manager.hpp"
+#include <iostream>
+#include <regex>
+#include <sstream>
+#include <string>
+
+std::string toString(Armor component) {
+    std::stringstream ss;
+
+    ss << "(" << component.armor << ")";
+    return ss.str();
+}
+
+std::string toString(Health component) {
+    std::stringstream ss;
+
+    ss << "(" << component.health << ")";
+    return ss.str();
+}
+std::string toString(Position component) {
+    std::stringstream ss;
+
+    ss << "(" << component.xpos << "," << component.ypos << ")";
+    return ss.str();
+}
+
+template <class T>
+std::string componentToString(EntityID entityId, Manager* manager) {
```

Comment:
**null_ptr** if entity does not have the specified component unhandled, this is a potential crash waiting to happen :bomb:

Reply:
Could be handled with no extra handling needed in entityToString() by returning empty string